src/_posts/gamestate.md

```diff
@@ -0,0 +1,346 @@
+6. The game state component publishes its state to the application by toggling
+the state components
+
+Below is a basic skeleton for game state system implementation that mirrors
+A Saturday Night. Also very thin in lines of code.
+
+```js
+AFRAME.registerComponent('game-state', {
+  schema: {
+    barCondition: {default: false},
+    state: {default: 'foo', oneOf: ['bar', 'qux', 'qaz']}
+  },
+
+  update: function (oldData) {
+    var data = this.data;
+    if (oldData.state !== data.state) {
+      this.setState(data.state);
+      return;
+    }
+    this.updateState();
+  },
+
+  updateState: function () {
+    var data = this.data;
+    var el = this.el;
+
+    switch (data.state) {
+      /**
+       * Application-specific code goes under here!
+       */
+      case 'foo': {
+        // Transition to next state depending on condition.
+        if (data.barCondition) {
+          el.setAttribute('game-state', 'state', 'bar');
+        }
+      }
+      default: {
+        console.log('Unknown state', data.state);
+      }
+    }
+  },
+
+  setState: function(state) {
+    var el = this.el;
+    var states = this.schema.state.oneOf;
+    if (this.updatingState) { return; }
+    if (states.indexOf(state) === -1) {
+      console.log('Unknown state', state);
+    }
+    // Remove all states.
+    states.forEach(function (state) {
+      el.removeAttribute(state);
+    });
+    el.setAttribute(state, '');
+  }
+});
+```
+
+Components can update the game state, and the game state will decide whether
+conditions have been fulfilled to transition to the next state.
+
+Then we have components that represent each distinct state:
+
+```js
+AFRAME.registerComponent('bar', {
+  init: function () {
+    // Ran when the game state enters the `bar` state.
+  },
+
+  remove: function () {
+    // Ran when the game state leaves the `bar` state.
+  }
+});
+```
+
+## Comparing Implementations Between A-Blast and A Saturday Night
+
+A-Blast's game state was something I came up with, basing it on Redux. The
+advantage is that components don't have to know about the game state, they just
+emit events, and the game state listens to them. A-Blast's game state is also
+more flexible since there's no single state represented as a string, any change
+to the game state represents a brand new game state. We just send actions, and
+the game state controls what needs to change in the state. The ideas from this
+state management has form of state management has been battle-tested through
+the web community's adoption of Redux.
+
+A Saturday Night's game state was implemented by Diego Marcos. It is well
+suited for simpler applications such as A Saturday Night where there are only a
+few very distinct states (avatar selection, dancing, sharing states). The nice
+part is that changing states and reacting to changed states are synchronous
+rather than through events. Though it would be possible to change A-Blast's
```

Comment:
last sentence is a dependent clause; you could start it like so: `Though it would be possible` -> `However, it would be possible`